gradle/dependencies.gradle

```diff
@@ -100,10 +100,10 @@ versions += [
   commonsCli: "1.4",
   commonsValidator: "1.7",
   dropwizardMetrics: "4.1.12.1",
-  gradle: "8.5",
+  gradle: "8.6",
```

Comment:
This PR has already been merged, but I'll leave a comment here ahyway.
I've noticed, what behaviour of incremental compilation for scala tests in gradle seems to be changed after gradle version upgrade from `8.5` to `8.6`.

With gradle `8.6`, if I do any change to some unit-test, e.g. `kafka.admin.AclCommandTest.scala`, and then run:

```
./gradlew core:compileTestScala
```

the entire folder `core/build/classes/scala/test` is cleared, and gradle re-compiles _all_ tests. It takes quite a long time. So the incremental compilation for scala tests seems to be broken.

If I change Gradle version back to `8.5`, incremental compilation works as expected

Reply:
That's a big deal, we should probably revert to the older version.

Reply:
https://github.com/apache/kafka/pull/15553 PR reverting to the old version.

@pasharik have you reported the bug (if it is indeed a bug in Gradle) to the upstream project?

Reply:
@jlprat my `trunk` branch was almost 2 weeks old, now I've updated it and I can't see any difference between Gradle 8.5 and 8.6. So I think we can leave this PR as it is, sorry for the confusion.

I didn't report issue to the Gradle yet, still trying to figure out exact steps how to reproduce it, and if it's related to Gradle or to my local environment setup.

Not related to this particular PR, but I still can see issues with incremental compilation, only when making changes from IntelliJ Idea. If changes are made from vim or other console text editor, incremental compilation works as expected.

Steps to reproduce:

* Open `kafka.admin.AclCommandTest` in IntelliJ Idea. I'm using `2023.3.5 Ultimate`
* Add `println("a")` at the beginning of `testAclCliWithAuthorizer` method
* Run in terminal: `./gradlew core:test --tests "kafka.admin.AclCommandTest.testAclCliWithAuthorizer"`
* Result: all tests in `core` module are re-compiled

If test modified with `vim`, only single test class is re-compiled